ee/middleware/upstreamoauth/oauth_client_utils.go

```diff
@@ -0,0 +1,46 @@
+package upstreamoauth
+
+import (
+	"net/http"
+
+	"github.com/TykTechnologies/tyk/internal/httpclient"
+)
+
+// createOAuthHTTPClient creates an HTTP client for OAuth operations with proper mTLS error handling
+func createOAuthHTTPClient(mw *Middleware) *http.Client {
+	gwConfig := mw.Gw.GetConfig()
+
+	// Check if external services are configured
+	if !gwConfig.ExternalServices.OAuth.MTLS.Enabled &&
```

Comment:
Question: Does this check happen if other places too? Do you think this can "grow" over time by adding more External Services?
If yes, maybe this check should have its own function so that we only need to maintain this check in one place, like `IsExternalServiceConfigured()`

Reply:
Thanks!